tst/EnergyPlus/unit/SolarShading.unit.cc

```diff
@@ -5534,21 +5510,378 @@ TEST_F(EnergyPlusFixture, SolarShadingTest_CalcInteriorSolarDistribution)
 
-    // HeatBalanceManager::GetZoneData(*state, FoundError); // Read Zone data from input file
-    // EXPECT_FALSE(FoundError);
+    //	compare_err_stream( "" ); // just for debugging
 
-    // SurfaceGeometry::GetGeometryParameters(*state, FoundError);
-    // EXPECT_FALSE(FoundError);
+    state->dataSurface->ShadingTransmittanceVaries = true;
+    state->dataSysVars->DetailedSkyDiffuseAlgorithm = true;
+    state->dataHeatBal->SolarDistribution = DataHeatBalance::Shadowing::FullExterior;
+
+    state->dataSolarShading->CalcSkyDifShading = true;
+    state->dataGlobal->BeginSimFlag = true;
+    HeatBalanceManager::InitHeatBalance(*state);
+    EnergyPlus::createFacilityElectricPowerServiceObject(*state);
+    HeatBalanceSurfaceManager::InitSurfaceHeatBalance(*state);
+    SolarShading::InitSolarCalculations(*state);
+
+    state->dataBSDFWindow->SUNCOSTS(4, 9)(1) = 0.1;
+    state->dataBSDFWindow->SUNCOSTS(4, 9)(2) = 0.1;
+    state->dataBSDFWindow->SUNCOSTS(4, 9)(3) = 0.1;
+    SolarShading::FigureSolarBeamAtTimestep(*state, state->dataGlobal->HourOfDay, state->dataGlobal->TimeStep);
+
+    int windowSurfNum = Util::FindItemInList("ZN001:WALL-SOUTH:WIN001", state->dataSurface->Surface);
+    int windowSurfNum2 = Util::FindItemInList("ZN001:FLOOR:WIN001", state->dataSurface->Surface);
+    EXPECT_NEAR(0.6504, state->dataSolarShading->SurfDifShdgRatioIsoSkyHRTS(4, 9, windowSurfNum), 0.0001);
+    EXPECT_NEAR(0.9152, state->dataSolarShading->SurfDifShdgRatioHorizHRTS(4, 9, windowSurfNum), 0.0001);
+
+    state->dataEnvrn->BeamSolarRad = 100.0;
+    state->dataHeatBal->SurfSunlitFrac(state->dataGlobal->HourOfDay, state->dataGlobal->TimeStep, windowSurfNum) = 1.0;
+    state->dataHeatBal->SurfCosIncAng(state->dataGlobal->HourOfDay, state->dataGlobal->TimeStep, windowSurfNum) = 1.0;
+    state->dataSurface->SurfWinGlTsolBmBm(windowSurfNum) = 0.8;
+    state->dataSurface->SurfWinGlTsolBmBm(windowSurfNum2) = 0.5;
+    state->dataSurface->SurfSkySolarInc(windowSurfNum) = 50.0;
+    state->dataSurface->SurfGndSolarInc(windowSurfNum) = 25.0;
 
+    SolarShading::CalcInteriorSolarDistribution(*state);
+    EXPECT_NEAR(1239.94, state->dataSurface->SurfWinTransSolar(windowSurfNum), 0.01);
+    EXPECT_NEAR(0.0, state->dataSurface->SurfWinTransSolar(windowSurfNum2), 0.01);
+    EXPECT_NEAR(0.646, state->dataSurface->SurfWinGlTsolBmBm(windowSurfNum), 0.01);
+    EXPECT_NEAR(0.0, state->dataSurface->SurfWinGlTsolBmBm(windowSurfNum2), 0.01);
+    EXPECT_NEAR(0.0, state->dataHeatBalSurf->SurfWinInitialBeamSolInTrans(windowSurfNum), 0.01);
+    EXPECT_NEAR(0.0, state->dataHeatBalSurf->SurfWinInitialBeamSolInTrans(windowSurfNum2), 0.01);
+    EXPECT_NEAR(0.0, state->dataHeatBalSurf->SurfWinInitialDifSolInTrans(windowSurfNum), 0.01);
+    EXPECT_NEAR(1.2, state->dataHeatBalSurf->SurfWinInitialDifSolInTrans(windowSurfNum2), 0.01);
+}
+TEST_F(EnergyPlusFixture, SolarShadingTest_CalcInteriorSolarDistribution_EQL)
+{
+    // Test for equivalent layer window model
```

Comment:
Clone existing unit test `SolarShadingTest_CalcInteriorSolarDistribution` and change window type to EquivalentLayer.

Reply:
Maybe it would have been nicer to not have a near duplicate, but it's fine really.